aiohttp/test_utils.py

```diff
@@ -490,7 +490,10 @@ def setup_test_loop(
     asyncio.set_event_loop(loop)
     if sys.platform != "win32" and not skip_watcher:
         policy = asyncio.get_event_loop_policy()
-        watcher = asyncio.SafeChildWatcher()
+        if hasattr(asyncio, "MultiLoopChildWatcher"):
+            watcher = asyncio.MultiLoopChildWatcher()
+        else:
+            watcher = asyncio.SafeChildWatcher()
```

Comment:
I have a few comments on this block: it is preferable to use the EAFP style in Python and also we need to have comments about the motivation of this.

To solve this, I suggest you cherry-pick my commit [`bea959a` (#5431)](https://github.com/aio-libs/aiohttp/pull/5431/commits/bea959a4ce195405939c40f184c5189206064447) (same patch: https://github.com/aio-libs/aiohttp/commit/bea959a4ce195405939c40f184c5189206064447.patch) and then change `ThreadedChildWatcher` with `MultiLoopChildWatcher` in a follow-up commit. If you know how to use interactive rebase in Git, it should be easy for you :)

Then, we could merge this change and backport it.

After that, I think it would be useful to vendor a backported watcher right in this repo and make use of it under older supported Pythons too. But this is a separate effort that should go into a separate PR.

Reply:
Thanks for the review.

I'm reading this ticket https://bugs.python.org/issue38591 and there seems to be many instabilities with `MultiLoopChildWatcher`. I think `ThreadedChildWatcher` might actually be a better choice here...

Regarding the performance concerns of creating a new thread, how about additionally exposing `skip_watcher: bool` as a param? `SafeChildWatcher` was originally introduced to support `asyncio.create_subprocess_*`. In my use case, this support is not needed. If we expose the param, no one pays the cost unnecessarily.

Reply:
Oh, maybe that's why I chose `ThreadedChildWatcher`. I think I talked to Andrew and he said to use it. It was still problematic under the heavy load on my laptop, IIRC.

As for `skip_watcher`, please file a proposal issue or a draft PR to demonstrate how you see it being exposed, API-wise.